demos/config.h

```diff
@@ -27,4 +27,19 @@ typedef int MQTTNetworkContext_t;
 
 #define MQTT_MAX_QUEUED_PUBLISH_MESSAGES 10
 
+/*
+ * @brief The Logging Interface Hook.
+ * 
+ * @note Customers can EITHER
+ * Define this macro to tie with their logging
+ * stack. 
+ *          OR
+ * Use the reference POSIX implementation, IotLog_Generic provided by us.
+ * 
+ * @param[in] messageLevel The integer logging level code for the message.
+ * @param[in] pFormat The format string for the log message.
+ * @param[in] ... The variadic argument list for the format string.
+ **/
+#define IotLog( messageLevel, pFormat, ... )
```

Comment:
Perhaps we can put:
```
#ifndef IotLog
    #define IotLog( messageLevel, pFormat, ...) /* Do nothing need to hook in. */
#endif
```
In every library header so that we can still build if missing from custom config.h.

Reply:
Made the change